lib/matplotlib/tests/test_ticker.py

```diff
@@ -1640,22 +1640,31 @@ def test_latex(self, is_latex, usetex, expected):
             assert fmt.format_pct(50, 100) == expected
 
 
-def test_locale_comma():
-    currentLocale = locale.getlocale()
+def _impl_locale_comma():
     try:
         locale.setlocale(locale.LC_ALL, 'de_DE.UTF-8')
-        ticks = mticker.ScalarFormatter(useMathText=True, useLocale=True)
-        fmt = '$\\mathdefault{%1.1f}$'
-        x = ticks._format_maybe_minus_and_locale(fmt, 0.5)
-        assert x == '$\\mathdefault{0{,}5}$'
-        # Do not change , in the format string
-        fmt = ',$\\mathdefault{,%1.1f},$'
-        x = ticks._format_maybe_minus_and_locale(fmt, 0.5)
-        assert x == ',$\\mathdefault{,0{,}5},$'
     except locale.Error:
-        pytest.skip("Locale de_DE.UTF-8 is not supported on this machine")
-    finally:
-        locale.setlocale(locale.LC_ALL, currentLocale)
+        print('SKIP: Locale de_DE.UTF-8 is not supported on this machine')
+        return
+    ticks = mticker.ScalarFormatter(useMathText=True, useLocale=True)
+    fmt = '$\\mathdefault{%1.1f}$'
+    x = ticks._format_maybe_minus_and_locale(fmt, 0.5)
+    assert x == '$\\mathdefault{0{,}5}$'
+    # Do not change , in the format string
+    fmt = ',$\\mathdefault{,%1.1f},$'
+    x = ticks._format_maybe_minus_and_locale(fmt, 0.5)
+    assert x == ',$\\mathdefault{,0{,}5},$'
+
+
+def test_locale_comma():
```

Comment:
Can you explain this song and dance?

Reply:
Yes, it's in the commit message:

> On some systems/pytest versions, the skip in an exception handler does not skip, but is treated as an exception. Namely, the ARM test machine in Cirrus and on my WSL Ubuntu.

I assume that is a bug somewhere in pytest but the subprocess avoids that. Also, changing the locale is a global modification, and it seems safer to do so in a subprocess, and not have to worry about fixing it up in the main process.

Reply:
Perhaps comment the code as well?  This seems an odd thing to do.

Reply:
Done.